src/base/scene.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+
+#include "camera.h"
+#include "mesh.h"
+#include "object.h"
+#include "tiny_gltf.h"
+
+namespace gm {
+class Scene {
+ public:
+  Scene(){};
+  Scene(const std::string &filepath);
+  void addObject(std::shared_ptr<SceneObject> o);
+  std::shared_ptr<PerspectiveCamera> getCamera();
+  std::vector<std::shared_ptr<SceneObject>> objects;
+
+ private:
+  std::shared_ptr<PerspectiveCamera> load_camera(tinygltf::Node node,
+                                                 tinygltf::Model model);
+  std::shared_ptr<Mesh> load_mesh(tinygltf::Node node, tinygltf::Model model);
+  std::shared_ptr<SceneObject> load_empty(tinygltf::Node node,
+                                          tinygltf::Model model);
+
+  std::vector<std::shared_ptr<SceneObject>> load_objects(
+      std::vector<int> node_ids, tinygltf::Model model,
+      std::shared_ptr<SceneObject> parent = nullptr);
```

Comment:
Nitpick: Do we need the default argument to be nullptr?

Reply:
Fixed in 3b8c543